src/lib/expression/aggregate_expression.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "abstract_expression.hpp"
+
+namespace opossum {
+
+enum class AggregateFunction { Min, Max, Sum, Avg, Count, CountDistinct };
+
+class AggregateExpression : public AbstractExpression {
+ public:
+  AggregateExpression(const AggregateFunction aggregate_function);
+  AggregateExpression(const AggregateFunction aggregate_function, const std::shared_ptr<AbstractExpression>& argument);
+
+  std::shared_ptr<AbstractExpression> argument() const;
+
+  std::shared_ptr<AbstractExpression> deep_copy() const override;
+  std::string as_column_name() const override;
+  DataType data_type() const override;
+
+  AggregateFunction aggregate_function;
```

Comment:
See above - should we really expose this as part of the class interface? Even more so if it is not const?

Reply:
on the list of things to discuss in person